netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -1,57 +1,104 @@
 package org.nlogo.app
 
 import java.awt.Rectangle
+import javax.swing.JComponent
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
 
 import org.nlogo.core.Widget
+import org.nlogo.window.MouseMode
+
+class WidgetUndoManager extends UndoManager{
```

Comment:
@Idloj Yes right now there is no use but I plann to use it in the future. The current state of code is very bad and I only pushed it so that @mrerrormessage can get a idea of what structure I am making.